triage/summarize.py

```diff
@@ -290,11 +287,53 @@ def tests_group_by_job(tests, builds):
                   key=lambda (k, v): (-len(v), k))
 
 
+SPAN_RE = re.compile(r'\w+|\W+')
+
+def common_spans(xs):
+    '''
+    Finds something similar to the longest common subsequence of xs, but much faster.
+
+    Returns a list of [matchlen_1, mismatchlen_2, matchlen_2, mismatchlen_2, ...], representing
+    sequences of the first element of the list that are present in all members.
+    '''
+    common = None
+    for x in xs:
+        x_split = SPAN_RE.findall(x)
+        if common is None:
```

Comment:
Consider noting that we want to distinguish first iteration from an empty intersection